packages/runtime/container-runtime/src/gc/gcSummaryStateTracker.ts

```diff
@@ -267,28 +266,23 @@ export class GCSummaryStateTracker {
 	 * Called to refresh the latest summary state. This happens when either a pending summary is acked or a snapshot
 	 * is downloaded and should be used to update the state.
 	 */
-	public async refreshLatestSummary(result: RefreshSummaryResult): Promise<void> {
-		// If the latest summary was updated and the summary was tracked, this client is the one that generated this
-		// summary. So, update wasGCRunInLatestSummary.
+	public async refreshLatestSummary(isSummaryTracked: boolean): Promise<void> {
+		// If the summary is tracked, this client is the one that generated it. So, update wasGCRunInLatestSummary.
 		// Note that this has to be updated if GC did not run too. Otherwise, `gcStateNeedsReset` will always return
 		// true in scenarios where GC is disabled but enabled in the snapshot we loaded from.
-		if (result.latestSummaryUpdated && result.wasSummaryTracked) {
+		if (isSummaryTracked) {
 			this.wasGCRunInLatestSummary = this.configs.shouldRunGC;
 		}
 
-		if (!result.latestSummaryUpdated || !this.configs.shouldRunGC) {
+		if (!isSummaryTracked || !this.configs.shouldRunGC) {
```

Comment:
nit: Instead of this early return, I think you can just nest an `if (this.configs.shouldRunGC)` inside the above if statement and put the code in there.  Might be clearer.  Or put this early return up top to reduce nesting.

Like this:

```ts
if (!isSummaryTracked) {
	return;
}

this.wasGCRunInLatestSummary = this.configs.shouldRunGC;

if (this.configs.shouldRunGC) {
	// The rest of the stuff
}
```

Reply:
Changed it. It looks pretty much the same TBH.